WFInfo/Data.cs

```diff
@@ -1257,60 +1259,147 @@ public async Task GetUserLogin(string email, string password)
             request.Dispose();
         }
 
+
+        // Some vibe-coded reflection modification for userAgent
+        public static void SetUserAgent(ClientWebSocketOptions options, string userAgent)
```

Comment:
I don't think this method actually works. Running it twice in a row, it falls into the `headers == null` block both times.

However, `options.Set` seems to allow the User-Agent header starting in .NET 5. I did a quick/dirty migration to .NET 9 to test and it seems to work, but would prefer confirmation from other side of connection.

Reply:
After some testing/discussion with Kyc, it's unclear if it actually works, and seems unlikely to be necessary. I don't think this is the detail that should make us upgrade from .NET Framework

Reply:
It does work. Without this method I get forbidden messages from Cloudflare.

Reply:
Kyc confirmed v1 websocket has some firewall rules and that's why current release is blocked, but also that v2 websocket doesn't and is unlikely to need any. 

If it empirically seemed to work, it must be something about how you tested it, not the actual code. 
- If you divide the chained statements into multiple steps, all the steps after `GetField` are null. 
- It's trying to treat the field as type `NameValueCollection`. The field is of type `WebHeaderCollection` which inherits from `NameValueCollection` so getting *could* work, but setting wouldn't.